src/miral/input_device_config.cpp

```diff
@@ -62,56 +78,187 @@ char const* const touchpad_click_mode_opt= "touchpad-click-mode";
+    }
+}
+
+auto convert_to_click_mode(std::optional<std::string> const& opt_val)
+-> std::optional<MirTouchpadClickMode>
+{
+    if (opt_val)
+    {
+        auto val = *opt_val;
+        if (val == touchpad_click_mode_none)
+        {
+            return mir_touchpad_click_mode_none;
+        }
+        else if (val == touchpad_click_mode_clickfinger)
+        {
+            return mir_touchpad_click_mode_finger_count;
+        }
+        else if (val == touchpad_click_mode_area)
+        {
+            return mir_touchpad_click_mode_area_to_click;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for {}: {}", touchpad_click_mode_opt, *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
+
+auto to_handedness(std::optional<std::string> const& opt_val)-> std::optional<MirPointerHandedness>
+{
+    if (opt_val)
+    {
+        if (*opt_val == right)
+        {
+            return mir_pointer_handedness_right;
+        }
+        else if (*opt_val == left)
+        {
+            return mir_pointer_handedness_left;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for handedness: {}", *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
 }
 
 void miral::add_input_device_configuration_options_to(mir::Server& server)
 {
-    server.add_configuration_option(mouse_acceleration_opt,
+    server.add_configuration_option(mouse_handedness_opt, std::format("Select mouse laterality: [{}, {}]", right, left),
+    mir::OptionType::string);
+    server.add_configuration_option(mouse_cursor_acceleration_opt,
                                     "Select acceleration profile for mice and trackballs [none, adaptive]",
                                     mir::OptionType::string);
     server.add_configuration_option(mouse_cursor_acceleration_bias_opt,
                                     "Constant factor (+1) to velocity or bias to the acceleration curve within the range [-1.0, 1.0] for mice",
                                     mir::OptionType::real);
-    server.add_configuration_option(mouse_scroll_speed_scale_opt,
-                                    "Scales mice scroll events, use negative values for natural scrolling",
+    server.add_configuration_option(mouse_scroll_speed_opt,
+                                    "Scales mouse scroll, use negative values for natural scrolling",
+                                    mir::OptionType::real);
+    server.add_configuration_option(mouse_hscroll_speed_override_opt,
+                                    "Scales mouse horizontal scroll, use negative values for natural scrolling",
+                                    mir::OptionType::real);
+    server.add_configuration_option(mouse_vscroll_speed_override_opt,
+                                    "Scales mouse vertical scroll, use negative values for natural scrolling",
```

Comment:
This seems like the two clauses here are sufficiently distinct that it'd be clearer to separate them into different sentences:
```suggestion
    server.add_configuration_option(mouse_scroll_speed_opt,
                                    "Scales mouse scroll. Use negative values for natural scrolling",
                                    mir::OptionType::real);
    server.add_configuration_option(mouse_hscroll_speed_override_opt,
                                    "Scales mouse horizontal scroll. Use negative values for natural scrolling",
                                    mir::OptionType::real);
    server.add_configuration_option(mouse_vscroll_speed_override_opt,
                                    "Scales mouse vertical scroll. Use negative values for natural scrolling",
```